barretenberg/cpp/src/barretenberg/vm/avm/recursion/avm_recursive_verifier.cpp

```diff
@@ -0,0 +1,95 @@
+#include "barretenberg/vm/avm/recursion/avm_recursive_verifier.hpp"
+#include "barretenberg/commitment_schemes/zeromorph/zeromorph.hpp"
+#include "barretenberg/transcript/transcript.hpp"
+
+namespace bb {
+
+template <typename Flavor>
+AvmRecursiveVerifier_<Flavor>::AvmRecursiveVerifier_(
+    Builder* builder, const std::shared_ptr<NativeVerificationKey>& native_verification_key)
+    : key(std::make_shared<VerificationKey>(builder, native_verification_key))
+    , builder(builder)
+{}
+
+template <typename Flavor>
+AvmRecursiveVerifier_<Flavor>::AvmRecursiveVerifier_(Builder* builder, const std::shared_ptr<VerificationKey>& vkey)
+    : key(vkey)
+    , builder(builder)
+{}
+
+template <typename Flavor>
+std::array<typename Flavor::GroupElement, 2> AvmRecursiveVerifier_<Flavor>::verify_proof(const HonkProof& proof)
```

Comment:
Would you please link this issue here https://github.com/AztecProtocol/barretenberg/issues/991 so we don't forget to change this pattern in the AvmRecursiveVerifier as well